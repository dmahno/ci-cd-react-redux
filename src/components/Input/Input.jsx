import React from 'react';
import './Input.scss';

function Input(props) {
  const {
    label,
    placeholder,
    required = 'free',
    type = 'text',
    width = 'input_width_normal',
    afterText = '',
    modifierFlex = 'flex_column',
    modifierInputTheme = 'input_theme_normal',
    onChange = () => {},
    hideClearButton = false,
    onReset = () => {},
    errorConnected = false,
    hasError = false,
    errorText = "",
    ...rest
  } = props;
  return (
    <div className={`input ${modifierInputTheme} ${modifierFlex}`}>
      <label className={`input__hint ${required}`}>{label}</label>
      <div className={`input__outer__wrapper`}>
      <div className={`input__inner__wrapper`}>
      <input
        type={type}
        className={`input__control input_size_m ${width}`}
        placeholder={placeholder}
        onChange={onChange}
        {...rest}
      />
      {
        hideClearButton? null: (<button type="button" className="input__clear" onClick={e => onReset()}><div>&times;</div></button>)
      }
      </div>
      <p className="error">{errorConnected && hasError? errorText: ""}</p>
      </div>
      <span className='input__after-text'>{afterText}</span>
    </div>
  );
}

export default Input;