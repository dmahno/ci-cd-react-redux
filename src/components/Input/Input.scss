.input__outer__wrapper {
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  .error {
    font-size: 12px;
    color: red;
    margin-left: 40px;
  }
  .input__inner__wrapper {
    width: 400px;
    position: relative;
    .input__clear {
      position: absolute;
      background-color: rgba(211, 211, 211, 0.637);
      // width: 5px;
      height: 15px;
      border-radius: 15px;
      border: none;
      padding-top: -10px;
      right: -5px;
      top: 10px;
      padding-bottom: 5px;
      bottom: 0;
      line-height: 40px;
      cursor: pointer;
      transition: 0.3s;
      outline: none;
      div {
        margin-top: -10px;
      }
    }
    .input__clear:hover {
      background-color: rgb(211, 211, 211);
    }
  }
}

.input__after-text {
  margin-left: -520px;
}


.input__control {
  border: 2px solid var(--color-bg-heavy);
  border-radius: 4px;
  color: var(--color-text);
  width: 100%;
  &:focus {
    border: 2px solid var(--color-borded-focused);
  }
}
//modifiers
.input_size_m {
  padding-left: 15px;
  height: 36px;
  font-size: 16px;
  max-width: 100%;
}
.input_theme_normal {
  padding: 8px 0;
}
.input_theme_big {
  padding: 12px 0;
}
.input_width_small {
  width: 30px;
  margin: 0 15px;
  text-align: right;
  padding: 0 10px;
}

//demands
.input-required {
  &:after {
    color: red;
    content: '*';
  }
}