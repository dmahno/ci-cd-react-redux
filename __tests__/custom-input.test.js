import renderer from "react-test-renderer";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import Input from '../src/components/Input/Input';

describe("Card component renders contents correctly", () => {
  it("renders correctly", () => {
    const inputData = {
      hideClearButton: true,
      label: "Enter the commit hash which you want to build",
      onBlur: e => () => {},
      onInput: e => () => {},
      onReset: e => () => {},
      placeholder: "Commit Hash",
      required: "input-required",
      value: ""
    };
    const rendered = renderer.create(
      <MemoryRouter>
        <Input {...inputData} />
      </MemoryRouter>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
