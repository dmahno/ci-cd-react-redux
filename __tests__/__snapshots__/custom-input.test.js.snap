// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders contents correctly renders correctly 1`] = `
<div
  className="input input_theme_normal flex_column"
>
  <label
    className="input__hint input-required"
  >
    Enter the commit hash which you want to build
  </label>
  <div
    className="input__outer__wrapper"
  >
    <div
      className="input__inner__wrapper"
    >
      <input
        className="input__control input_size_m input_width_normal"
        onBlur={[Function]}
        onChange={[Function]}
        onInput={[Function]}
        placeholder="Commit Hash"
        type="text"
        value=""
      />
    </div>
    <p
      className="error"
    >
      
    </p>
  </div>
  <span
    className="input__after-text"
  >
    
  </span>
</div>
`;
